server/src/test/java/org/opensearch/cluster/metadata/QueryGroupTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.common.ResourceType;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.test.AbstractSerializingTestCase;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class QueryGroupTests extends AbstractSerializingTestCase<QueryGroup> {
```

Comment:
Can we write more tests for `toXContent` and `fromXContent`?

Reply:
We can write the `toXContent` method testCase and `fromXContent` testCase is already implemented in `AbstractSerializingTestCase ` class